core/src/main/java/org/arquillian/smart/testing/scm/git/GitChangeResolver.java

```diff
@@ -62,10 +62,10 @@ public GitChangeResolver(File dir, String previous, String head) {
         final FileRepositoryBuilder builder = new FileRepositoryBuilder();
         try {
             this.git = new Git(builder.readEnvironment().findGitDir(dir).build());
-        } catch (IOException e) {
-            throw new IllegalArgumentException("Unable to find git repository for path " + dir.getAbsolutePath(), e);
+            this.repoRoot = git.getRepository().getDirectory().getParentFile();
+        } catch (IllegalArgumentException | IOException e) {
```

Comment:
Isn't there a better way than just catching the exception?
eg: `builder.readEnvironment().findGitDir(dir).getGitDir() != null`